### PR TITLE
Adding a box on the webboot menu to keep track of logs

### PR DESCRIPTION
**Pull request by pgherghetta**

### CODE REVIEW

File: pkg/menu/menu.go [hunk 1]
```diff
@@ -494,4 +532,9 @@ func (p *Progress) Close() {
 		<-p.ackTerm
 	}
 	ui.Clear()
+	//func (p *Progress) Update(text string) {
```
Comment (rjoleary) [inline pkg/menu/menu.go:535]: Cleanup the unused code.

File: pkg/wifi/iwl.go [hunk 2]
```diff
@@ -167,16 +174,25 @@ func (w *IWLWorker) Connect(stdout, stderr io.Writer, a ...string) error {
 
 	// There's no telling how long the supplicant will take, but on the other hand,
 	// it's been almost instantaneous. But, further, it needs to keep running.
+	//log.Print("Hello 2")
```
Comment (rjoleary) [inline pkg/wifi/iwl.go:177]: Cleanup debug prints.

File: pkg/menu/menu.go [hunk 3]
```diff
@@ -362,22 +372,50 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
...
+	location += alternateHeight
 	input := newParagraph("", true, location, windowWidth, height)
 
+	location += height
+	logBox := widgets.NewList()
+	logBox.Title = "Logs:"
+	logBox.WrapText = false
+	logBox.SetRect(0, location, windowWidth, location+height)
+	logBox.Rows = []string{
+		"",
```
Comment (rjoleary) [inline pkg/menu/menu.go:393]: Does the empty string have a purpose here?

**Review by rjoleary — COMMENTED** [review]
Squash the commits together.

File: pkg/menu/menu.go [hunk 4]
```diff
@@ -362,22 +372,50 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
...
 	warning := newParagraph("<Esc> to go back, <Ctrl+d> to exit", false, location, windowWidth, height)
+	// Write the contents of the log output text file to the log box.
+	var file, err = os.OpenFile("logOutput.txt", os.O_APPEND|os.O_CREATE|os.O_RDWR, 0755)
+	if err != nil {
+		log.Fatal(err)
+	}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		for i := 0; i < len(logBox.Rows); i++ {
+			logBox.Rows[i] += scanner.Text()
```
Comment (rjoleary) [inline pkg/menu/menu.go:406]: Add a newline at the end of this string.

File: pkg/menu/menu.go [hunk 5]
```diff
@@ -354,6 +364,12 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
 		listData = append(listData, fmt.Sprintf("[%d] %s", i, e.Label()))
 	}
 	windowWidth, windowHeight := termbox.Size()
+
+	if err := ui.Init(); err != nil {
```
Comment (rjoleary) [inline pkg/menu/menu.go:368]: Try removing this block.

File: pkg/menu/menu.go [hunk 6]
```diff
@@ -362,22 +378,92 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
...
+		"8",
+		"9",
+		"10",
+		"11",
+		"12",
+		"13",
+		"14",
+		"15",
+	}
+	uiEvents = ui.PollEvents()
```
Comment (rjoleary) [inline pkg/menu/menu.go:416]: Remove this line.

File: pkg/menu/menu.go [hunk 7]
```diff
@@ -362,22 +378,92 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
...
 	input := newParagraph("", true, location, windowWidth, height)
 
+	location += height
+	logBox := widgets.NewList()
+	logBox.Title = "Logs:"
+	logBox.WrapText = false
+	logBox.SetRect(0, location, windowWidth, location+height)
+	logBox.Rows = []string{}
+
+	logBox.Rows = []string{
```
Comment (rjoleary) [inline pkg/menu/menu.go:398]: You don't need this since `logBox.Rows` was set just above.

File: pkg/wifi/iwl.go [hunk 8]
```diff
@@ -168,15 +175,18 @@ func (w *IWLWorker) Connect(stdout, stderr io.Writer, a ...string) error {
...
+		outfile, err := os.OpenFile("logOutput.txt", os.O_APPEND|os.O_CREATE|os.O_RDWR, 0755)
 		cmd.Stdout, cmd.Stderr = outfile, outfile
 		if err != nil {
 			log.Print(err)
 		}
+		defer outfile.Close()
 		if err = cmd.Run(); err != nil {
 			c <- err
+			outfile.WriteString(time.Now().String() + " ")
+			outfile.WriteString(err.Error())
```
Comment (rjoleary) [inline pkg/wifi/iwl.go:189]: Add a newline after this line.

File: pkg/menu/menu.go [hunk 9]
```diff
@@ -362,22 +378,92 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
...
+		log.Fatal(err)
+	}
 
-	ui.Render(intro)
 	ui.Render(input)
 	ui.Render(warning)
+	ui.Render(logBox)
 
-	chooseIndex, err := parsingMenuOption(listData, menu, input, warning, uiEvents, customWarning...)
+	for {
```
Comment (rjoleary) [inline pkg/menu/menu.go:456]: Remove this for loop.

File: pkg/menu/menu.go [hunk 10]
```diff
@@ -362,22 +372,64 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
...
+	location += alternateHeight
 	input := newParagraph("", true, location, windowWidth, height)
 
+	location += height
+	logBox := widgets.NewList()
+	logBox.Title = "Logs:"
+	logBox.WrapText = false
+	logBox.SetRect(0, location, windowWidth, location+height)
+	logBox.Rows = []string{}
+	logBox.Rows = []string{"1",
```
Comment (rjoleary) [inline pkg/menu/menu.go:393]: You can remove these numbers -- they were just for debugging.

File: pkg/wifi/iwl.go [hunk 11]
```diff
@@ -157,6 +157,14 @@ func (w *IWLWorker) Connect(stdout, stderr io.Writer, a ...string) error {
 	}
 
 	if err := ioutil.WriteFile("/tmp/wifi.conf", conf, 0444); err != nil {
+		var file, err = os.OpenFile("logOutput.txt", os.O_APPEND|os.O_CREATE|os.O_RDWR, 0755)
+		if err != nil {
+			return err
+		}
+		file.WriteString(time.Now().String() + " ")
+		file.WriteString("/temp/wifi.conf: " + err.Error())
+		file.WriteString("\n")
```
Comment (kellysun878) [inline pkg/wifi/iwl.go:166]: Maybe use fmt.Sprintf here.

File: pkg/menu/menu.go [hunk 12]
```diff
@@ -362,22 +374,50 @@ func PromptMenuEntry(menuTitle string, introwords string, entries []Entry, uiEve
...
+	logBox.Title = "Logs:"
+	logBox.WrapText = false
+	logBox.SetRect(0, location, windowWidth, location+height)
+	logBox.Rows = make([]string, 1000)
+
 	location += height
 	warning := newParagraph("<Esc> to go back, <Ctrl+d> to exit", false, location, windowWidth, height)
 
+	// Write the contents of the log output text file to the log box.
+	var file, err = os.OpenFile("logOutput.txt", os.O_APPEND|os.O_CREATE|os.O_RDWR, 0755)
```
Comment (rjoleary) [inline pkg/menu/menu.go:400]: 0644

File: pkg/wifi/iwl.go [hunk 13]
```diff
@@ -168,15 +176,19 @@ func (w *IWLWorker) Connect(stdout, stderr io.Writer, a ...string) error {
 	// There's no telling how long the supplicant will take, but on the other hand,
 	// it's been almost instantaneous. But, further, it needs to keep running.
 	go func() {
-		cmd := exec.CommandContext(ctx, "/usr/bin/strace", "-f", "-o", "/tmp/out", "-s", "1024", "-v", "wpa_supplicant", "-i"+w.Interface, "-c/tmp/wifi.conf", "-f", "wpa_supplicant.log", "-dd")
+		cmd := exec.CommandContext(ctx, "/usr/bin/strace", "-f", "-o", "/tmp/out", "-s", "1024", "-v", "wpa_supplicant", "-i"+w.Interface, "-c/tmp/wifi.conf", "-dd")
 
-		outfile, err := os.OpenFile("morelog", os.O_WRONLY|os.O_CREATE, 0666)
+		outfile, err := os.OpenFile("logOutput.txt", os.O_APPEND|os.O_CREATE|os.O_RDWR, 0755)
```
Comment (rminnich) [inline pkg/wifi/iwl.go:181]: 0755 -> 0644

Comment (rminnich) [reply pkg/wifi/iwl.go:181]: os.O_WRONLY

File: pkg/wifi/iwl.go (hunk 11 above)
Comment (rminnich) [reply pkg/wifi/iwl.go:166]: os.O_WRONLY not os.O_RDWR since you will only write it.

**Review by rminnich — COMMENTED** [review]
just about there.

**Review by rjoleary — APPROVED** [review]

**Pull request merged**